microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -168,6 +169,37 @@ def _check_rx_or_tx_pps(
                 "than 2^20 (~1m) PPS after sriov disable."
             ).is_less_than(2 ** 20)
 
+    @TestCaseMetadata(
+        description="""
+            verify vpp is able to detect azure network interfaces
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_nic_count=2,
+            network_interface=Sriov(),
+        ),
+    )
+    def verify_dpdk_vpp(
+        self, node: Node, log: Logger, variables: Dict[str, Any]
+    ) -> None:
+
+        vpp = DpdkVpp(node)
+        vpp.install()
```

Comment:
What's the reason not to follow the standard way to get the tool? The installation will be triggered automatically. If write in this way, the tool may be installed multiple times on the same node in some running scenarios.

Reply:
I was just following the same way we did this with Testpmd, since we keep the tool in the local test directory I don't think the node.tools[Tool] way works.

Reply:
Sorry, I missed this in previews reviews. Calling by node.tools doesn't affect where the tool is installed. It helps to manage the lifecycle of the tool.

Reply:
I'll raise a second PR to switch both of them to the correct call format, does that work?

Reply:
sure

Reply:
nevermind I'll just do it here, making lots of changes anyway